### PR TITLE
Build framework structure

### PR DESCRIPTION
**Pull request by Phoeneix**
Github Issue: #[22](https://github.com/podaac/hydrocron/issues/22)

### Description

Emulate users querying the API from public internet and characterize performance of the API.
- Create separate repo for testing.
- Set up the framework

### Overview of work done

    - Created CHANGELOG and Version file
    - Created test starter
    - Created config files
    - Created Hydrocon API Interface
    - Created basic file handling methods
    - Created Ping test
    - Created dependencies file for pip
    - Added multiple thread handling to code
    - Set up logging
    - Added locust

### Overview of verification done

Local execution
image

## PR checklist:

* [x] Updated changelog
* [x] Executed on UAT

_See [Pull Request Review Checklist](../CONTRIBUTING.md#reviewing) for pointers on reviewing this pull request_

### CODE REVIEW

**Review by frankinspace — COMMENTED** [review]
Did you look for standard/public libraries for API load testing in python? I found this one which looks like it could simplify a lot of this code and give us more features:
https://locust.io/

**Comment by Phoeneix** [comment]
> Did you look for standard/public libraries for API load testing in python? I found this one which looks like it could simplify a lot of this code and give us more features: https://locust.io/

I did look around for solutions for parallel testing. What I found is already incorporated into the framework. Also based on what was requested, I was making API tests, not UI tests.

**Comment by Phoeneix** [comment]
Updated framework to use Locust

**Review by frankinspace — COMMENTED** [review]
If you use the same project structure as our other python libraries and use poetry for the build it will simplify the cicd because we can reuse the same structure we use for those projects as well.

File: dependencies_pip
Comment (frankinspace) [inline dependencies_pip:1]: Need to add locust as dependency

File: utils/setupcalls.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import subprocess, os, sys, platform
+
+
+class SetupCalls():
+
+    def SetUpLogging():
+        print(f"\r\nSetting up file logging...")
+        tee = subprocess.Popen(["tee", "log_LatestRun.txt"], stdin=subprocess.PIPE)
```
Comment (frankinspace) [inline utils/setupcalls.py:8]: Is there a reason we need this instead of just using the `logging` module?

File: executeApiTests.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    threadCount = _args.thread_count
+
+    SetupCalls.SetUpLogging()
+    ConfigManager.InitializeConfig(
+        env = environment,
+        threadCount = threadCount)
+    argumentList = ["tests", "--junitxml=./test-results/report.xml", '-v']
+    # pytest.main(argumentList)
+    
+    # Set up paralel execution
```
Comment (frankinspace) [inline executeApiTests.py:62]: Do we still need this if locust is handling threading?

File: utils/setupcalls.py (hunk 1 above)
Comment (Phoeneix) [reply utils/setupcalls.py:8]: I did this for pytest originally to put the log in a file for debugging, but the locust log is not useful for that. It's not removed yet just like the other pytest code. I was planning to keep it until the test development as a fallback case if needed.

File: executeApiTests.py (hunk 2 above)
Comment (Phoeneix) [reply executeApiTests.py:62]: I was planning to keep it until the test development as a fallback case if I encounter a situation that locust can't handle and I can't find a workaround for it.

File: dependencies_pip
Comment (Phoeneix) [reply dependencies_pip:1]: Updated dependencies

**Comment by Phoeneix** [comment]
> If you use the same project structure as our other python libraries and use poetry for the build it will simplify the cicd because we can reuse the same structure we use for those projects as well.

Sure, can be included in the CI/CD implementation ticket

**Review by frankinspace — APPROVED** [review]

**Pull request merged**